com.regnosys.rosetta.tests/src/com/regnosys/rosetta/generator/java/function/FunctionGeneratorTest.xtend

```diff
@@ -35,6 +35,55 @@ class FunctionGeneratorTest {
 	@Inject extension ModelHelper
 	@Inject extension ValidationTestHelper
 	
+	@Test
+	def void directlyUseAttributesOfImplicitVariableTest() {
+		val code = '''
+			namespace com.rosetta.test.model
+			version "${project.version}"
+			
+			type Foo:
+				a int (1..1)
+				b string (0..*)
+			
+			func F1:
+				inputs:
+					foos Foo (0..*)
+				output:
+					result int (0..*)
+				
+				add result:
+					foos
+						extract [ a + b count]
+		'''.generateCode
+		val classes = code.compileToClasses
+
+		val foo1 = classes.createInstanceUsingBuilder('Foo', of('a', 42, 'b', #[]))
+		val foo2 = classes.createInstanceUsingBuilder('Foo', of('a', -5, 'b', #["Hello", "World!"]))
+		val func1 = classes.createFunc("F1");
+		assertEquals(#[42, -3], func1.invokeFunc(List, #[#[foo1, foo2]]))
+	}
+	
+	@Test
+	def void omittedParameterInFunctionalOperationTest() {
+		val code = '''
+			namespace com.rosetta.test.model
+			version "${project.version}"
+			
+			func F1:
+				inputs:
+					a int (0..*)
+				output:
+					result int (0..*)
+				
+				add result:
+					a extract [+ 1]
```

Comment:
The left argument of _every_ operation (`extract`, `sum`, and yes, even `+`) is now implicitly filled with `item` when left out.